etna/pipeline/base.py

```diff
@@ -673,6 +675,10 @@ def _compute_metrics(
         self, metrics: List[Metric], y_true: TSDataset, y_pred: TSDataset
     ) -> Dict[str, Dict[str, float]]:
         """Compute metrics for given y_true, y_pred."""
+        if y_true.has_hierarchy():
+            if y_true.current_df_level != y_pred.current_df_level:
+                y_true = y_true.get_level_dataset(y_pred.current_df_level)  # type: ignore
```

Comment:
Why is there any problem with typing?

Reply:
Mypy think that `y_pred. current_df_level` might be `None`,  in fact if `y_true` has hierarchical structure then `y_pred` should too